### ISSUE TITLE
Integrate findbugs

### ISSUE THREAD

**Issue by hssm** [ankidroid/Anki-Android#1266]
Originally reported on Google Code with ID 363

```
Findbugs (http://findbugs.sourceforge.net/) is a code checker using static analysis.
It can be used to discover common programming mistakes, it would make a great addition
to improve code quality.

An android specific plugin for findbugs is already in the works and will be released
soon.

Tasks:

 * add ant task to run findbugs
 * create a config file
 * squash bugs!

```

Reported by `jan.berkel` on 2011-01-29 11:48:26

**Comment by mikehardy** [comment]
I looked into this and there are a huge number of static analyzers, and several ways to integrate them (developer IDE, build system, hosted analyzers in a CI system), and several ways to take action (hard fail on certain classes of problem, only accept a decreasing number of problems, just generate a report, auto-log issues, etc etc).

Given IDE integration is so easy (it's just a couple clicks these days, I've done it) I don't think this issue is well-defined. Further, AnkiDroid has such an incredibly noisy profile for most code smell issues that this would require an enormous effort to clean up, and I've not experienced that happening in previous projects with similar profiles.

TL;DR - @timrae I just don't see this as actionable though I do have a playground branch that integrates sonarcloud.io (hosted static analysis that integrates with Travis CI). https://sonarcloud.io/organizations/mikehardy-github/projects 

If that's interesting I'll make a PR to do the integration than propose something like "CI fails if blocker+critical count increments", or if that isn't interesting than I have no idea how this issue would actually be closable...
